### PR TITLE
update composer removing the dependency on cakephp/cakephp and to dev…

### LINKED ISSUE #8: Seems dependency is only on cake orm
Hi @josegonzalez , 

I am notified about this repo by lorenzo . Looking at the repo, I believe this have dependency on the cake/orm rather than on the cake framework itself.

If you think so, I wished this could be a library which can be used by other people who are using cake as a standalone version.

Thank you for considering.

### PR DESCRIPTION
**Pull request by harikt**
…. Current time cannot remove the cake dependency due to no tests in orm resides

Fixes #8 partially.

Once https://github.com/cakephp/cakephp/issues/6994 is fixed , can make the necessary adjustments.

Thank you.

### CODE REVIEW

**Comment by harikt** [comment]
Some of the tests are broken dude to `satooshi/php-coveralls dev-master requires php >=5.5 -> your PHP version (5.4.37) or value of "config.platform.php" in composer.json does not satisfy that requirement.` .

**Comment by josegonzalez** [comment]
I dropped 5.4 support on master. Good riddance.

Can you rebase onto master?

File: composer.json [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {
     "name": "josegonzalez/cakephp-version",
-    "description": "CakePHP plugin to allow versioning of records",
-    "type": "cakephp-plugin",
-    "keywords": ["cakephp", "revision", "version"],
+    "description": "Cake orm behavior to allow versioning of records",
+    "type": "library",
+    "keywords": ["cake", "orm", "revision", "version"],
```
Comment (josegonzalez) [inline composer.json]: why did you change the keyword from cakephp to cake?

File: composer.json [hunk 2]
```diff
@@ -1,8 +1,8 @@
 {
     "name": "josegonzalez/cakephp-version",
-    "description": "CakePHP plugin to allow versioning of records",
-    "type": "cakephp-plugin",
-    "keywords": ["cakephp", "revision", "version"],
+    "description": "Cake orm behavior to allow versioning of records",
```
Comment (josegonzalez) [inline composer.json]: ORM should be capitalized.

File: composer.json [hunk 3]
```diff
@@ -1,8 +1,8 @@
...
+    "type": "library",
```
Comment (josegonzalez) [inline composer.json]: Please do not change this. Does this fail to install in non-cakephp applications without this change?

File: composer.json (hunk 1 above)
Comment (harikt) [inline composer.json]: I may be wrong, I thought cakephp seems more like a framework. May be add cakephp also .

File: composer.json (hunk 2 above)
Comment (harikt) [inline composer.json]: sure.

File: composer.json (hunk 3 above)
Comment (harikt) [inline composer.json]: this was what I was referring in the comment https://github.com/josegonzalez/cakephp-version/issues/8#issuecomment-171860536 .

I am wondering if you have a type of `cakephp-plugin` how will it then handle the same? I will try and see how it will work.

**Comment by harikt** [comment]
@josegonzalez made the changes and rebased.

Let me know if you think there is any other issues.

Thank you.

**Comment by harikt** [comment]
For the record, you are correct that without the change in the type to `cakephp-plugin` it will work.

Thank you.
